lib/pos_blockchain.go

```diff
@@ -633,6 +643,27 @@ func (bc *Blockchain) hasValidBlockViewPoS(block *MsgDeSoBlock) error {
 	return nil
 }
 
+func (bc *Blockchain) hasValidProposerRandomSeedSignaturePoS(block *MsgDeSoBlock) (bool, error) {
+	// Validate that the leader proposed a valid random seed signature.
+	parentBlock, exists := bc.blockIndexByHash[*block.Header.PrevBlockHash]
+	if !exists {
+		// Note: this should never happen as we only call this function after
+		// we've validated that all ancestors exist in the block index.
+		return false, RuleErrorMissingParentBlock
+	}
+
+	prevRandomSeedHashSHA256 := sha256.Sum256(parentBlock.Header.ProposerRandomSeedSignature.ToBytes())
+	prevRandomSeedHash, err := (&RandomSeedHash{}).FromBytes(prevRandomSeedHashSHA256[:])
```

Comment:
You can reuse this new function once you define it below
```suggestion
	prevRandomSeedHash := hashRandomSeedSignature(parentBlock.Header.ProposerRandomSeedSignature)
```